libenv/unix_iface.c

```diff
@@ -752,4 +755,370 @@ static int aix_get_mac_addr(const char *device_name, uint8_t mac[6])
+
+    GetProcFileInfo(ctx, "/proc/net/route",  "routes", NULL, &NetworkingRoutesPostProcessInfo,
+                    // format: Iface	Destination	Gateway 	Flags	RefCnt	Use	Metric	Mask		MTU	Window	IRTT
+                    //         eth0	00000000	0102A8C0	0003	0	0	1024	00000000	0	0	0 
+                    "^(?<interface>\\S+)\\t(?<raw_dest>[[:xdigit:]]+)\\t(?<raw_gw>[[:xdigit:]]+)\\t(?<raw_flags>[[:xdigit:]]+)\\t(?<refcnt>\\d+)\\t(?<use>\\d+)\\t(?<metric>[[:xdigit:]]+)\\t(?<raw_mask>[[:xdigit:]]+)\\t(?<mtu>\\d+)\\t(?<window>\\d+)\\t(?<irtt>[[:xdigit:]]+)");
+
+    GetProcFileInfo(ctx, "/proc/net/ipv6_route",  "ipv6_routes", NULL, &NetworkingIPv6RoutesPostProcessInfo,
+                    // format: dest                    dest_prefix source                source_prefix next_hop                         metric   refcnt   use      flags        interface
+                    //         fe800000000000000000000000000000 40 00000000000000000000000000000000 00 00000000000000000000000000000000 00000100 00000000 00000000 00000001     eth0
+                    "^(?<raw_dest>[[:xdigit:]]+)\\s+(?<dest_prefix>[[:xdigit:]]+)\\s+"
+                    "(?<raw_source>[[:xdigit:]]+)\\s+(?<source_prefix>[[:xdigit:]]+)\\s+"
+                    "(?<raw_next_hop>[[:xdigit:]]+)\\s+(?<raw_metric>[[:xdigit:]]+)\\s+"
+                    "(?<refcnt>\\d+)\\s+(?<use>\\d+)\\s+"
+                    "(?<raw_flags>[[:xdigit:]]+)\\s+(?<interface>\\S+)");
+
+    const char* ports_regex = "^\\s*\\d+:\\s+(?<raw_local>[0-9A-F:]+)\\s+(?<raw_remote>[0-9A-F:]+)\\s+(?<raw_state>[0-9]+)";
+
+    GetProcFileInfo(ctx, "/proc/net/tcp",  "ports_tcp", NULL, &NetworkingPortsPostProcessInfo, ports_regex);
+    GetProcFileInfo(ctx, "/proc/net/tcp6", "ports_tcp6", NULL, &NetworkingPortsPostProcessInfo, ports_regex);
+    GetProcFileInfo(ctx, "/proc/net/udp",  "ports_udp", NULL, &NetworkingPortsPostProcessInfo, ports_regex);
+    GetProcFileInfo(ctx, "/proc/net/udp6", "ports_udp6", NULL, &NetworkingPortsPostProcessInfo, ports_regex);
+
+    // Inter-|   Receive                                                |  Transmit
+    //  face |bytes    packets errs drop fifo frame compressed multicast|bytes    packets errs drop fifo colls carrier compressed
+    //   eth0: 74850544807 75236137    0    0    0     0          0   1108775 63111535625 74696758    0    0    0     0       0          0
+
+    GetProcFileInfo(ctx, "/proc/net/dev", "dev", "device", NULL,
+                    "^\\s*(?<device>[^:]+)\\s*:\\s*"
+                    "(?<receive_bytes>\\d+)\\s+(?<receive_packets>\\d+)\\s+(?<receive_errors>\\d+)\\s+(?<receive_drop>\\d+)\\s+(?<receive_fifo>\\d+)\\s+(?<receive_frame>\\d+)\\s+(?<receive_compressed>\\d+)\\s+(?<receive_multicast>\\d+)\\s+"
+                    "(?<transmit_bytes>\\d+)\\s+(?<transmit_packets>\\d+)\\s+(?<transmit_errors>\\d+)\\s+(?<transmit_drop>\\d+)\\s+(?<transmit_fifo>\\d+)\\s+(?<transmit_frame>\\d+)\\s+(?<transmit_compressed>\\d+)\\s+(?<transmit_multicast>\\d+)");
+}
```

Comment:
+1, nice way of doing it with `GetProcFileInfo()`.

I'd break up the lines a little bit to make it easier to read, like you already did for `ipv6_routes`.

I won't even attempt to decipher those regexes. :-)